### PR TITLE
Add an optional seed parameter in Colour.random

### PR DESCRIPTION
**Pull request by Maselkov**
## Summary

Adds an optional seed parameter in Color.Random
Example use case: using a user's id to generate a consistent, random embed color without the need to store it.

## Checklist

- [X] If code changes were made then they have been tested.
    - [X] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [X] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/colour.py [hunk 1]
```diff
@@ -119,17 +119,25 @@ def default(cls):
...
         .. note::
 
             The random algorithm works by choosing a colour with a random hue but
             with maxed out saturation and value.
 
+        Parameters
+        ------------
+        seed: Optional[Union[:class:`int`, :class:`str`, :class:`float`, :class:`bytes`, :class:`bytearray`]]
+            The seed to initialize the colour with.
+
```
Comment (SebbyLaw) [inline discord/colour.py:134]: Parameters should be below the 1.6 versionadded marker

Should also have a 1.7 versionadded under the seed parameter

**Review by SebbyLaw — CHANGES_REQUESTED** [review]

File: discord/colour.py (hunk 1 above)
Comment (Bluenix2) [reply discord/colour.py:136]: No? Should it not be a function-wide `versionchanged`?

Comment (ShineyDev) [reply discord/colour.py:136]: The [documentation](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged) implicitly states that `.. versionchanged::` should be used here, but I've seen both used in this context throughout the library - though I do also recall a consistency PR changing this.

**Comment by Rapptz** [comment]
I'm not sure I really get the point of this. The use case seems really weak.

**Comment by Maselkov** [comment]
I mostly added it because I generate groups of related roles based on a parameter and having a single, randomized color for the whole group is convenient. The use case presented is just another one I quickly came up with. You can always generate the color by yourself, but I think this is a nice shortcut.

File: discord/colour.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def default(cls):
         return cls(0)
 
     @classmethod
-    def random(cls):
+    def random(cls, seed=None):
```
Comment (Rapptz) [inline discord/colour.py:122]:
```suggestion
    def random(cls, *, seed=None):
```

File: discord/colour.py [hunk 3]
```diff
@@ -128,8 +128,18 @@ def random(cls):
             with maxed out saturation and value.
 
         .. versionadded:: 1.6
+
+        Parameters
+        ------------
+        seed: Optional[Union[:class:`int`, :class:`str`, :class:`float`, :class:`bytes`, :class:`bytearray`]]
+            The seed to initialize the colour with.
```
Comment (Rapptz) [inline discord/colour.py:135]:
```suggestion
            The seed to initialize the RNG with. If ``None`` is passed the default RNG is used. 
```

File: discord/colour.py [hunk 4]
```diff
@@ -128,8 +128,18 @@ def random(cls):
...
+
+            .. versionadded:: 1.7
         """
-        return cls.from_hsv(random.random(), 1, 1)
+        rand = random
+        if seed:
+            rand = random.Random(seed)
```
Comment (Rapptz) [inline discord/colour.py:141]:
```suggestion
        rand = random if seed is None else random.Random(seed)
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

**Comment by Midorina** [comment]
I don't really get the point of this. Exact color int or a seed, you store something either way. Could you provide an example use case where this addition will be useful?

**Comment by pikaninja** [comment]
They left one in the pr's description?
> Example use case: using a user's id to generate a consistent, random embed color without the need to store it.
You can't give such a large number as the color int

**Comment by Midorina** [comment]
@pikaninja oh, makes sense, thanks

**Comment by Vexs** [comment]
I brought this up in the discord server but it's worth actually publishing I guess- you don't need to use seeding to get a random color here, you can just use `id % 255` to get a random-enough color.

**Comment by Maselkov** [comment]
> you can just use `id % 255` to get a random-enough color.
That would lead to a very limited color range :p. And to be fair, my own use case involves strings being the seed rather than ids. The use case I presented above is something I thought might be somewhat more common

Anyway, I won't argue whether this change is worth adding or not. My viewpoint might be skewed and maybe it is indeed too niche to be worth merging. Still, I think that it could be useful from time to time.

**Comment by Vexs** [comment]
color.random() uses HSV- the hue is the only thing randomly selected. I thought HSV was 0-255 per, but it's 0-1. `id%255^3` would give you the whole range of rgb, but as HSV with SV maxed is better, you'd want something like:
`color.from_hsv((id%1000)/1000, 1, 1)` (this isn't quite perfect, but w/e). 
As for handling strings, you could just `sum()` the `ord()`, or bitshift them around, etc- and then take that number and mod it.

**Review by Rapptz — APPROVED** [review]
The code is fine now, thanks.

I'm still not really sure if I understand the use cases but I guess there might be something I'm missing. It's fine though, thanks for the PR.

**Pull request merged**